src/components/VueSearchbox.jsx

```diff
@@ -0,0 +1,111 @@
+import Input from "../styles/Input";
+import Searchbase from "@appbaseio/searchbase";
+
+const VueSearchbox = {
+  name: "VueSearchbox", // vue component name
+  props: {
+    app: types.app,
+    url: types.url,
+    credentials: types.credentials,
+    analytics: types.analytics,
+    headers: types.headers,
+    dataField: types.dataField,
+    nestedField: types.nestedField,
+    title: types.title,
+    defaultValue: types.defaultValue,
+    value: types.value,
+    placeholder: types.placeholder,
+    showIcon: types.showIcon,
+    iconPosition: types.iconPosition,
+    icon: types.icon,
+    showClear: types.showClear,
+    clearIcon: types.clearIcon,
+    autosuggest: types.autosuggest,
+    strictSelection: types.strictSelection,
+    defaultSuggestions: types.defaultSuggestions,
+    debounce: types.debounce,
+    highlight: types.highlight,
+    highlightField: types.highlightField,
+    customHighlight: types.customHighlight,
+    queryFormat: types.queryFormat,
+    fuzziness: types.fuzziness,
+    showVoiceSearch: types.showVoiceSearch,
+    searchOperators: types.searchOperators,
+    render: types.render,
+    renderError: types.renderError,
+    renderNoSuggestion: types.renderNoSuggestion,
+    getMicInstance: types.getMicInstance,
+    renderMic: types.renderMic,
+    onChange: types.onChange,
+    onValueChange: types.onValueChange,
+    onSuggestions: types.onSuggestions,
+    onError: types.onError,
+    onResults: types.onResults,
+    innerClass: types.innerClass,
+    defaultQuery: types.defaultQuery,
+    beforeValueChange: types.beforeValueChange,
+    onQueryChange: types.onQueryChange,
+    className: types.className,
+    loader: types.loader,
+    onBlur: types.onBlur,
+    onKeyPress: types.onKeyPress,
+    onKeyUp: types.onKeyUp,
+    onFocus: types.onFocus,
+    onKeyDown: types.onKeyDown,
+    autoFocus: types.autoFocus
+  },
+  created() {
+    this._initSearchBase();
+  },
+  methods: {
+    _initSearchBase() {
+      const {
+        app,
+        url,
+        dataField,
+        credentials,
+        analytics,
+        headers,
+        nestedField,
+        defaultQuery,
+        beforeValueChange,
+        queryFormat,
+        defaultSuggestions,
+        fuzziness,
+        searchOperators
+      } = this.$props;
+
+      try {
+        const transformQuery = query => {
+          if (defaultQuery) return defaultQuery(query, this.state.currentValue);
+          return Promise.resolve(query);
+        };
+
+        this.searchBase = new Searchbase({
+          index: app,
+          url,
+          dataField,
+          credentials,
+          analytics,
+          headers,
+          nestedField,
+          transformQuery,
+          beforeValueChange,
+          queryFormat,
+          suggestions: defaultSuggestions,
+          fuzziness,
+          searchOperators
+        });
+      } catch (e) {
+        console.error(e);
+      }
+    }
+  },
+  render() {
+    console.log(this.searchBase);
```

Comment:
Lets remove this

Reply:
kept it deliberately for debugging, all `console.log` will be removed in the final PR